### PR TITLE
Feature/dynamic column #7

### PR DESCRIPTION
**Pull request by Noamagnus**
#7 

https://i.gyazo.com/c846218bc6d33749b44a614a45bb0bbf.mp4

### CODE REVIEW

**Review by Egle0111 — APPROVED** [review]

**Review by MrgaCRO — APPROVED** [review]
Kod mene radi sve kako si i prikazao.

File: lib/ui/molecule/resize_widget.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  @override
+  _ResizeWidgetState createState() => _ResizeWidgetState();
+}
+
+double containerHeight = 40;
+
+class _ResizeWidgetState extends State<ResizeWidget> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: scaffold and appbar are not needed in organism widget

File: lib/ui/molecule/resize_widget.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                  child: Text(
+                    'Red content',
+                    style: TextStyle(
+                      fontSize: 25,
+                    ),
+                  )),
+              decoration: BoxDecoration(color: Colors.red),
+            ),
+          ),
+          Divider(
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: gesture detector needs to wrap divider, and not bottom container

Comment (matejthetree) [reply lib/ui/molecule/resize_widget.dart]: you can make thickness/height a 3

Comment (matejthetree) [reply lib/ui/molecule/resize_widget.dart]: thikness

Comment (matejthetree) [reply lib/ui/molecule/resize_widget.dart]: or, wrap gesture detector around complete column, and then check if the (dx,dy) coordinates are close to the divider.

File: lib/ui/molecule/resize_widget.dart [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+
+class ResizeWidget extends StatefulWidget {
+  @override
+  _ResizeWidgetState createState() => _ResizeWidgetState();
+}
+
+double containerHeight = 40;
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: this needs to be inside state

File: lib/ui/molecule/resize_widget.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+class _ResizeWidgetState extends State<ResizeWidget> {
+  @override
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart:29]: add container height here

File: lib/ui/molecule/resize_widget.dart [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+
+class ResizeWidget extends StatefulWidget {
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart:3]:
```
final Widget topChild;
final Widget bottomChild;
final double dividerPercentage;

... constructor
```

File: lib/ui/molecule/resize_widget.dart [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('neki text'),
+      ),
+      body: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: <Widget>[
+          Expanded(
+            child: Container(
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: widget.topChild

File: lib/ui/molecule/resize_widget.dart [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                  if (containerHeight >= 40) {
+                    containerHeight = containerHeight - details.delta.dy;
+                  }
+                }
+                print('container height $containerHeight');
+                print('delta ${details.delta}');
+                print('local ${details.localPosition}');
+              });
+            },
+            child: Container(
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: `widget.bottomChild`

File: lib/ui/molecule/resize_widget.dart (hunk 4 above)
Comment (matejthetree) [reply lib/ui/molecule/resize_widget.dart:29]:
```
double _currentPercentage;
initState .... ->
 _currentPercentage = widget.dividerPercentage;
```

then inside build function, you will use this percentage to calculate container height from MediaQuery size

Comment (matejthetree) [reply lib/ui/molecule/resize_widget.dart:29]:
```
Column...
[
Expanded(flex: _currentPercantage*100),
Divider(),
Expanded(flex: (1-_currentPercantage)*100)
```

**Review by matejthetree — CHANGES_REQUESTED** [review]

File: lib/ui/molecule/resize_widget.dart [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+              _currentPercentage = widget.dividerPercentage;
+            }
+            if (_currentPercentage < _upperBorder) {
+              _currentPercentage = _upperBorder;
+            }
+            // ovaj deo code-a mi sluzi da bih shvatio valicine a i video kako se one menjaju
+            print('current% $_currentPercentage'); //
+            print('divider% ${widget.dividerPercentage}');
+//
+//            print('global ${details.globalPosition.dy}');
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: pobrisi komentare

File: lib/ui/molecule/resize_widget.dart [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            }
+            //resenje problema sa decimalama krkan style
+            if (_currentPercentage > widget.dividerPercentage) {
+              _currentPercentage = widget.dividerPercentage;
+            }
+            if (_currentPercentage < _upperBorder) {
+              _currentPercentage = _upperBorder;
+            }
+            // ovaj deo code-a mi sluzi da bih shvatio valicine a i video kako se one menjaju
+            print('current% $_currentPercentage'); //
```
Comment (matejthetree) [inline lib/ui/molecule/resize_widget.dart]: ovo isto pobrisi ako si resio, neznam imamo li logger implementiran. za sada brisi.

**Review by matejthetree — APPROVED** [review]
super. pobrisi komentare i mozemo mergat

**Pull request merged**